pandas/io/formats/style.py

```diff
@@ -1871,17 +1871,17 @@ def apply_index(
         >>> df = pd.DataFrame([[1,2], [3,4]], index=["A", "B"])
         >>> def color_b(s):
         ...     return {ret}
-        >>> df.style.{this}_index(color_b)  # doctest: +SKIP
+        >>> df.style.apply_index(color_b)  # doctest: +SKIP
```

Comment:
This is a bit more complex than other docstrings, since it's a template that is reused in different docstrings, and things like `{this}` are variables that are replaced by values before the docstring is being used.

See for example how `{this}` is not displayed in the final documentation, but a vaue is replaced with: https://pandas.pydata.org/docs/dev/reference/api/pandas.io.formats.style.Styler.apply_index.html

I don't think those should be a problem, since the doctests should run on the rendered docstring and not the template. Can you revert and see if something else is causing the error. Maybe I'm wrong, and the problem is that we run the doctests in the docstring. If that's the case, let's better revert the changes, and open a separate issue for that.

If you want to see what's the rendered docstring, you should be able to get it with something like `help(pandas.DataFrame.style.apply_index)`.

Reply:
I looked into this issue further and here are my findings: 

Using the ```help``` function, I confirmed that the docstrings are correctly rendered. 

So I created a test file under pandas, using the ```@doc``` decorator to test how the template strings are being tested under flake8. 
![image](https://user-images.githubusercontent.com/66008784/218749185-22b6f965-c569-45aa-b02b-1808a7ce5146.png)
![image](https://user-images.githubusercontent.com/66008784/218749263-77caccab-4971-4c2f-89c1-b919e7d2f29b.png)

From the screenshot it seems like when these template strings are in the first line of the doctest i.e. line starting with ```>>>```, it is flagged as a syntax error. The succeeding lines are not flagged as syntax errors, explaining why the other parts of the doctest using string interpolation such as ```return {ret}``` are OK. 

My guess is that perhaps flake8 identifies the doctests chunks & runs the syntax checking before the docstrings are rendered, and there is perhaps a bug (or perhaps there are other reasons) where the string interpolation is flagged as a syntax error for the first line. 

I tried to search if others were experiencing similar issues online, but was unable to find more information. 

@datapythonista Let me know your thoughts!

Reply:
Thanks a lot for the detailed research and explanation @kkangs0226. I find it surprising that pytest would extract the docstrings statically instead of just running the code and using the `__doc__` attribute. But your test shows that it's exactly what's happening.

As @MarcoGorelli suggests, I think better to revert the changes related to template variables for now, and maybe open an issue with the problem. So, we can merge your other changes here, and have a discussion on what we want to do with the template problem. If you create the issue, feel free to get the permalink to your comment above and reference it in the issue, since it provides a very clear explanation of the problem and what's going on.